File: generators/latinDiseases.py
__all__ = ['latinDiseases']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['nm1', 'nm2', 'nm3', 'result', 'nm4'])
    var.put('nm1', Js([Js([Js('Abscessus'), Js('Abscess')]), Js([Js('Accessio'), Js('Seizure')]), Js([Js('Agonia'), Js('Cramps')]), Js([Js('Ambustio'), Js('Burn')]), Js([Js('Aneurisma'), Js('Aneurysm')]), Js([Js('Angina'), Js('Chest pain')]), Js([Js('Apoplexia'), Js('Stroke')]), Js([Js('Asthenia'), Js('Atrophy')]), Js([Js('Ataxia'), Js('Ataxia')]), Js([Js('Atrophia'), Js('Atrophy')]), Js([Js('Calculus'), Js('Stones')]), Js([Js('Cancrum'), Js('Canker')]), Js([Js('Carbunculus'), Js('Carbuncle')]), Js([Js('Carcinoma'), Js('Cancer')]), Js([Js('Catarrhus'), Js('Catarrh')]), Js([Js('Cholerica'), Js('Cholera')]), Js([Js('Chorea'), Js('Dance')]), Js([Js('Colica'), Js('Colic')]), Js([Js('Commotio'), Js('Concussion')]), Js([Js('Contracture'), Js('Stricture')]), Js([Js('Contuses'), Js('Bruised')]), Js([Js('Contusio'), Js('Contusion')]), Js([Js('Convulsio'), Js('Convulsion')]), Js([Js('Convulsionis'), Js('Convulsions')]), Js([Js('Crampus'), Js('Cramps')]), Js([Js('Debilitas'), Js('Weakness')]), Js([Js('Dolor'), Js('Pain')]), Js([Js('Ecclampsia'), Js('Convulsions')]), Js([Js('Exhaustio'), Js('Exhaustion')]), Js([Js('Explosio'), Js('Explosion')]), Js([Js('Fatuitas'), Js('Idiocy')]), Js([Js('Febris'), Js('Fever')]), Js([Js('Fracture'), Js('Fracture')]), Js([Js('Fulmen'), Js('Lightning')]), Js([Js('Galbanus'), Js('Jaundice')]), Js([Js('Gangraena'), Js('Gangrene')]), Js([Js('Gelatio'), Js('Frost')]), Js([Js('Glarea'), Js('Gravel')]), Js([Js('Haemorrhagia'), Js('Hemorrhage')]), Js([Js('Haemorrhois'), Js('Hemorrhoids')]), Js([Js('Hydrocephalus'), Js('Hydrocephalus')]), Js([Js('Hydropisis'), Js('Dropsis')]), Js([Js('Hydrops'), Js('Dropsis')]), Js([Js('Ictus'), Js('Trauma')]), Js([Js('Ignis'), Js('Burning')]), Js([Js('Infectio'), Js('Infection')]), Js([Js('Infirmus'), Js('Weak')]), Js([Js('Inflammatio'), Js('Inflammation')]), Js([Js('Intessusceptio'), Js('Intussusception')]), Js([Js('Lepra'), Js('Leprosy')]), Js([Js('Marasmus'), Js('Weakness')]), Js([Js('Mollities'), Js('Softening')]), Js([Js('Morbilli'), Js('Measles')]), Js([Js('Morbus'), Js('Disease')]), Js([Js('Mors'), Js('Death')]), Js([Js('Myelitis'), Js('Paraplegia')]), Js([Js('Noctis'), Js('of the Night')]), Js([Js('Noma'), Js('Canker')]), Js([Js('Obitus'), Js('Death')]), Js([Js('Obstructiones'), Js('Obstruction')]), Js([Js('Otitis'), Js('Inflammation')]), Js([Js('Peritus'), Js('Deceased')]), Js([Js('Perniciose'), Js('Pernicious')]), Js([Js('Pestis'), Js('Plague')]), Js([Js('Phthisis'), Js('Consumption')]), Js([Js('Plaga'), Js('Plague')]), Js([Js('Pleuritis'), Js('Pleuritis')]), Js([Js('Privatio'), Js('Priviation')]), Js([Js('Rheumatismus'), Js('Rheumatism')]), Js([Js('Scophulosis'), Js('Scrofula')]), Js([Js('Scorbutus'), Js('Scurvy')]), Js([Js('Senilis'), Js('Weakness')]), Js([Js('Spasmus'), Js('Spasms')]), Js([Js('Tussis'), Js('Cough')]), Js([Js('Ulcus'), Js('Ulcer')]), Js([Js('Venenatio'), Js('Poisoning')]), Js([Js('Vermis'), Js('Worms')]), Js([Js('Vitium'), Js('Disease')])]))
    var.put('nm2', Js([Js([Js('Abdmonis'), Js('Abdominal')]), Js([Js('Abdominalis'), Js('Abdominal')]), Js([Js('Acerbus'), Js('Sharp')]), Js([Js('Agita'), Js('Shaking')]), Js([Js('Alvus'), Js('Belly')]), Js([Js('Articulorum'), Js('Joint')]), Js([Js('Bracium'), Js('Arm')]), Js([Js('Caput'), Js('Head')]), Js([Js('Cerebralis'), Js('Cerebrum')]), Js([Js('Cerebri'), Js('Brain')]), Js([Js('Cerebri'), Js('Cerebral')]), Js([Js('Clarus'), Js('Clear')]), Js([Js('Collum'), Js('Neck')]), Js([Js('Cordis'), Js('Heart')]), Js([Js('Coxa'), Js('Hip')]), Js([Js('Cutis'), Js('Skin')]), Js([Js('Dentes'), Js('Teeth')]), Js([Js('Dextra'), Js('Right')]), Js([Js('Epiglottidis'), Js('Eppiglotis')]), Js([Js('Flava'), Js('Yellow')]), Js([Js('Flores'), Js('Flower')]), Js([Js('Folia'), Js('Leaf')]), Js([Js('Frigor'), Js('Cold')]), Js([Js('Fuscus'), Js('Brown')]), Js([Js('Fuscus'), Js('Dark')]), Js([Js('Glottidis'), Js('Glottis')]), Js([Js('Hectica'), Js('Hectic')]), Js([Js('Hirudo'), Js('Leech')]), Js([Js('Ilii'), Js('Ilium')]), Js([Js('Incisum'), Js('Cut')]), Js([Js('Incognita'), Js('Unidentified')]), Js([Js('Infantilis'), Js('Infantile')]), Js([Js('Inflammatoria'), Js('Inflammatory')]), Js([Js('Intermittens'), Js('Intermittent')]), Js([Js('Jecoris'), Js('Liver')]), Js([Js('Laceratum'), Js('Lacerated')]), Js([Js('Laryngea'), Js('Larynx')]), Js([Js('Luteus'), Js('Yellow')]), Js([Js('Magnus'), Js('Mighty')]), Js([Js('Membranacea'), Js('Membrane')]), Js([Js('Motus'), Js('Locomotion')]), Js([Js('Nervosa'), Js('Nervous')]), Js([Js('Niger'), Js('Black')]), Js([Js('Molle'), Js('Soft')]), Js([Js('Mollis'), Js('Soft')]), Js([Js('Os'), Js('Mouth')]), Js([Js('Ovarii'), Js('Ovarian')]), Js([Js('Pancreatis'), Js('Pancreas')]), Js([Js('Pectus'), Js('Breast')]), Js([Js('Pedicularis'), Js('Louse')]), Js([Js('Pes'), Js('Foot')]), Js([Js('Petechialis'), Js('Spotted')]), Js([Js('Prostata'), Js('Prostate')]), Js([Js('Puerperalis'), Js('Puerperal')]), Js([Js('Pulmonum'), Js('Lung')]), Js([Js('Punctum'), Js('Stabbed')]), Js([Js('Putrida'), Js('Rotten')]), Js([Js('Remittens'), Js('Remittent')]), Js([Js('Rheumatica'), Js('Rheumatic')]), Js([Js('Rubra'), Js('Scarlet')]), Js([Js('Rumen'), Js('Throat')]), Js([Js('Siccus'), Js('Dry')]), Js([Js('Spissus'), Js('Dense')]), Js([Js('Sacer'), Js('Sacred')]), Js([Js('Senilis'), Js('Dry')]), Js([Js('Vulnus'), Js('Wounded')]), Js([Js('Ustus'), Js('Burnt')]), Js([Js('Sinistra'), Js('Left')]), Js([Js('Tonsillaris'), Js('Tonsils')]), Js([Js('Ulna'), Js('Arm')]), Js([Js('Ulna'), Js('Elbow')]), Js([Js('Uteri'), Js('Uterus')]), Js([Js('Ventriculi'), Js('Stomach')]), Js([Js('Verminosa'), Js('Verminous')]), Js([Js('Astuosa'), Js('Hot')]), Js([Js('Aquaticum'), Js('Aquatic')]), Js([Js('Arenosum'), Js('Sandy')]), Js([Js('Candidulum'), Js('Shining')]), Js([Js('Cavum'), Js('Hollow')]), Js([Js('Cerritum'), Js('Frantic')])]))
    var.put('nm3', Js([Js('Ab'), Js('Absce'), Js('Acce'), Js('Ago'), Js('Am'), Js('Ambu'), Js('Aneu'), Js('Angi'), Js('Apo'), Js('As'), Js('Asthe'), Js('Atro'), Js('Ca'), Js('Cal'), Js('Calcu'), Js('Can'), Js('Car'), Js('Carbu'), Js('Cata'), Js('Cho'), Js('Chole'), Js('Co'), Js('Coli'), Js('Commo'), Js('Con'), Js('Contra'), Js('Contu'), Js('Convu'), Js('Cra'), Js('Cram'), Js('Debi'), Js('Do'), Js('Eccla'), Js('Ex'), Js('Explo'), Js('Fa'), Js('Fatui'), Js('Fe'), Js('Fi'), Js('Fra'), Js('Ga'), Js('Gal'), Js('Galba'), Js('Gela'), Js('Gla'), Js('Haemo'), Js('Hy'), Js('Hydro'), Js('In'), Js('Infe'), Js('InfiInfla'), Js('Inte'), Js('Intes'), Js('Le'), Js('Ma'), Js('Mara'), Js('Molli'), Js('Mor'), Js('Morbi'), Js('Mye'), Js('Noc'), Js('Peri'), Js('Perni'), Js('Pes'), Js('Phthi'), Js('Pla'), Js('Pleu'), Js('Pri'), Js('Priva'), Js('Rheu'), Js('Sco'), Js('Scophu'), Js('Scor'), Js('Se'), Js('Seni'), Js('Spa'), Js('Spas'), Js('Tu'), Js('Ve'), Js('Vene'), Js('Ver'), Js('Vi')]))
    var.put('nm4', Js([Js('banus'), Js('bus'), Js('cinoma'), Js('ciose'), Js('crum'), Js('culus'), Js('cus'), Js('drops'), Js('fectio'), Js('firmus'), Js('ga'), Js('gina'), Js('graena'), Js('hagia'), Js('li'), Js('lica'), Js('lis'), Js('litas'), Js('lities'), Js('litis'), Js('lor'), Js('losis'), Js('lus'), Js('ma'), Js('matio'), Js('men'), Js('mis'), Js('motio'), Js('mus'), Js('natio'), Js('nia'), Js('nis'), Js('noma'), Js('nus'), Js('phalus'), Js('phia'), Js('pisis'), Js('plexia'), Js('pus'), Js('ra'), Js('rea'), Js('rhagia'), Js('rhus'), Js('rica'), Js('ris'), Js('risma'), Js('ritis'), Js('sceptio'), Js('ses'), Js('sia'), Js('sio'), Js('sionis'), Js('sis'), Js('sus'), Js('tas'), Js('ties'), Js('tio'), Js('tiones'), Js('tis'), Js('tismus'), Js('tium'), Js('ture'), Js('tus'), Js('tuses'), Js('vatio'), Js('xia')]))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if (var.get('i')<Js(5.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                var.put('names', (((((((var.get('nm1').get(var.get('rnd')).get('0')+Js(' '))+var.get('nm2').get(var.get('rnd2')).get('0'))+Js(' ('))+var.get('nm2').get(var.get('rnd2')).get('1'))+Js(' '))+var.get('nm1').get(var.get('rnd')).get('1'))+Js(')')))
                var.get('nm1').callprop('splice', var.get('rnd'), Js(1.0))
                var.get('nm2').callprop('splice', var.get('rnd2'), Js(1.0))
            else:
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                while PyJsStrictEq(var.get('rnd'),var.get('rnd3')):
                    var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                while PyJsStrictEq(var.get('rnd2'),var.get('rnd4')):
                    var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                var.put('names', ((((var.get('nm3').get(var.get('rnd'))+var.get('nm4').get(var.get('rnd2')))+Js(' '))+var.get('nm3').get(var.get('rnd3')))+var.get('nm4').get(var.get('rnd4'))))
                var.get('nm3').callprop('splice', var.get('rnd'), Js(1.0))
                var.get('nm3').callprop('splice', var.get('rnd3'), Js(1.0))
                var.get('nm4').callprop('splice', var.get('rnd2'), Js(1.0))
                var.get('nm4').callprop('splice', var.get('rnd4'), Js(1.0))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
pass
pass


# Add lib to the module scope
latinDiseases = var.to_python()

File: js/latinDiseases.js

function nameGen(){
	var nm1 = [["Abscessus","Abscess"],["Accessio","Seizure"],["Agonia","Cramps"],["Ambustio","Burn"],["Aneurisma","Aneurysm"],["Angina","Chest pain"],["Apoplexia","Stroke"],["Asthenia","Atrophy"],["Ataxia","Ataxia"],["Atrophia","Atrophy"],["Calculus","Stones"],["Cancrum","Canker"],["Carbunculus","Carbuncle"],["Carcinoma","Cancer"],["Catarrhus","Catarrh"],["Cholerica","Cholera"],["Chorea","Dance"],["Colica","Colic"],["Commotio","Concussion"],["Contracture","Stricture"],["Contuses","Bruised"],["Contusio","Contusion"],["Convulsio","Convulsion"],["Convulsionis","Convulsions"],["Crampus","Cramps"],["Debilitas","Weakness"],["Dolor","Pain"],["Ecclampsia","Convulsions"],["Exhaustio","Exhaustion"],["Explosio","Explosion"],["Fatuitas","Idiocy"],["Febris","Fever"],["Fracture","Fracture"],["Fulmen","Lightning"],["Galbanus","Jaundice"],["Gangraena","Gangrene"],["Gelatio","Frost"],["Glarea","Gravel"],["Haemorrhagia","Hemorrhage"],["Haemorrhois","Hemorrhoids"],["Hydrocephalus","Hydrocephalus"],["Hydropisis","Dropsis"],["Hydrops","Dropsis"],["Ictus","Trauma"],["Ignis","Burning"],["Infectio","Infection"],["Infirmus","Weak"],["Inflammatio","Inflammation"],["Intessusceptio","Intussusception"],["Lepra","Leprosy"],["Marasmus","Weakness"],["Mollities","Softening"],["Morbilli","Measles"],["Morbus","Disease"],["Mors","Death"],["Myelitis","Paraplegia"],["Noctis","of the Night"],["Noma","Canker"],["Obitus","Death"],["Obstructiones","Obstruction"],["Otitis","Inflammation"],["Peritus","Deceased"],["Perniciose","Pernicious"],["Pestis","Plague"],["Phthisis","Consumption"],["Plaga","Plague"],["Pleuritis","Pleuritis"],["Privatio","Priviation"],["Rheumatismus","Rheumatism"],["Scophulosis","Scrofula"],["Scorbutus","Scurvy"],["Senilis","Weakness"],["Spasmus","Spasms"],["Tussis","Cough"],["Ulcus","Ulcer"],["Venenatio","Poisoning"],["Vermis","Worms"],["Vitium","Disease"]];
	var nm2 = [["Abdmonis","Abdominal"],["Abdominalis","Abdominal"],["Acerbus","Sharp"],["Agita","Shaking"],["Alvus","Belly"],["Articulorum","Joint"],["Bracium","Arm"],["Caput","Head"],["Cerebralis","Cerebrum"],["Cerebri","Brain"],["Cerebri","Cerebral"],["Clarus","Clear"],["Collum","Neck"],["Cordis","Heart"],["Coxa","Hip"],["Cutis","Skin"],["Dentes","Teeth"],["Dextra","Right"],["Epiglottidis","Eppiglotis"],["Flava","Yellow"],["Flores","Flower"],["Folia","Leaf"],["Frigor","Cold"],["Fuscus","Brown"],["Fuscus","Dark"],["Glottidis","Glottis"],["Hectica","Hectic"],["Hirudo","Leech"],["Ilii","Ilium"],["Incisum","Cut"],["Incognita","Unidentified"],["Infantilis","Infantile"],["Inflammatoria","Inflammatory"],["Intermittens","Intermittent"],["Jecoris","Liver"],["Laceratum","Lacerated"],["Laryngea","Larynx"],["Luteus","Yellow"],["Magnus","Mighty"],["Membranacea","Membrane"],["Motus","Locomotion"],["Nervosa","Nervous"],["Niger","Black"],["Molle","Soft"],["Mollis","Soft"],["Os","Mouth"],["Ovarii","Ovarian"],["Pancreatis","Pancreas"],["Pectus","Breast"],["Pedicularis","Louse"],["Pes","Foot"],["Petechialis","Spotted"],["Prostata","Prostate"],["Puerperalis","Puerperal"],["Pulmonum","Lung"],["Punctum","Stabbed"],["Putrida","Rotten"],["Remittens","Remittent"],["Rheumatica","Rheumatic"],["Rubra","Scarlet"],["Rumen","Throat"],["Siccus","Dry"],["Spissus","Dense"],["Sacer","Sacred"],["Senilis","Dry"],["Vulnus","Wounded"],["Ustus","Burnt"],["Sinistra","Left"],["Tonsillaris","Tonsils"],["Ulna","Arm"],["Ulna","Elbow"],["Uteri","Uterus"],["Ventriculi","Stomach"],["Verminosa","Verminous"],["Astuosa","Hot"],["Aquaticum","Aquatic"],["Arenosum","Sandy"],["Candidulum","Shining"],["Cavum","Hollow"],["Cerritum","Frantic"]];
	var nm3 = ["Ab","Absce","Acce","Ago","Am","Ambu","Aneu","Angi","Apo","As","Asthe","Atro","Ca","Cal","Calcu","Can","Car","Carbu","Cata","Cho","Chole","Co","Coli","Commo","Con","Contra","Contu","Convu","Cra","Cram","Debi","Do","Eccla","Ex","Explo","Fa","Fatui","Fe","Fi","Fra","Ga","Gal","Galba","Gela","Gla","Haemo","Hy","Hydro","In","Infe","InfiInfla","Inte","Intes","Le","Ma","Mara","Molli","Mor","Morbi","Mye","Noc","Peri","Perni","Pes","Phthi","Pla","Pleu","Pri","Priva","Rheu","Sco","Scophu","Scor","Se","Seni","Spa","Spas","Tu","Ve","Vene","Ver","Vi"];
	var nm4 = ["banus","bus","cinoma","ciose","crum","culus","cus","drops","fectio","firmus","ga","gina","graena","hagia","li","lica","lis","litas","lities","litis","lor","losis","lus","ma","matio","men","mis","motio","mus","natio","nia","nis","noma","nus","phalus","phia","pisis","plexia","pus","ra","rea","rhagia","rhus","rica","ris","risma","ritis","sceptio","ses","sia","sio","sionis","sis","sus","tas","ties","tio","tiones","tis","tismus","tium","ture","tus","tuses","vatio","xia"];
	
	var br = "";

	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(i < 5){
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			names = nm1[rnd][0] + " " + nm2[rnd2][0] + " ("  + nm2[rnd2][1] + " " + nm1[rnd][1] + ")";
			nm1.splice(rnd, 1);
			nm2.splice(rnd2, 1);
		}else{
			rnd = Math.floor(Math.random() * nm3.length);
			rnd2 = Math.floor(Math.random() * nm4.length);
			rnd3 = Math.floor(Math.random() * nm3.length);
			while(rnd === rnd3){
				rnd3 = Math.floor(Math.random() * nm3.length);
			}
			rnd4 = Math.floor(Math.random() * nm4.length);
			while(rnd2 === rnd4){
				rnd4 = Math.floor(Math.random() * nm4.length);
			}
			names = nm3[rnd] + nm4[rnd2] + " " + nm3[rnd3] + nm4[rnd4];
			nm3.splice(rnd, 1);
			nm3.splice(rnd3, 1);
			nm4.splice(rnd2, 1);
			nm4.splice(rnd4, 1);
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}